src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembership.java

```diff
@@ -3,55 +3,54 @@
 import uk.gov.companieshouse.api.acsp_manage_users.api.UserAcspMembershipInterface;
 import uk.gov.companieshouse.api.acsp_manage_users.model.AcspMembership;
 import uk.gov.companieshouse.api.acsp_manage_users.model.RequestBodyPatch;
 import uk.gov.companieshouse.api.acsp_manage_users.model.RequestBodyPost;
 import uk.gov.companieshouse.api.acsp_manage_users.model.ResponseBodyPost;
 
-import java.util.List;
-
 @RestController
 public class UserAcspMembership implements UserAcspMembershipInterface {
 
-    public UserAcspMembership() {}
-
-    @Override
-    public ResponseEntity<ResponseBodyPost> addAcspMember(
-            @NotNull String xRequestId,
-            @NotNull String ericIdentity,
-            @Valid RequestBodyPost requestBodyPost
-    ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1149)
-    }
-
-    @Override
-    public ResponseEntity<AcspMembership> getAcspMembershipForAcspId(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[a-zA-Z0-9]*$") String id
-    ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1146)
-    }
-
-    @Override
-    public ResponseEntity<List<AcspMembership>> getAcspMembershipForUserId(
-            @NotNull String xRequestId,
-            @NotNull String ericIdentity,
-            @Valid Boolean includeRemoved
-    ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1145)
-    }
-
-    @Override
-    public ResponseEntity<Void> updateAcspMembershipForId(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[a-zA-Z0-9]*$") String id,
-            @Valid RequestBodyPatch requestBodyPatch
-    ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1147)
-    }
-
+  private final AcspMembersService acspMembersService;
+
+  public UserAcspMembership(final AcspMembersService acspMembersService) {
+    this.acspMembersService = acspMembersService;
+  }
+
+  @Override
+  public ResponseEntity<ResponseBodyPost> addAcspMember(
+      @NotNull String xRequestId,
+      @NotNull String ericIdentity,
+      @Valid RequestBodyPost requestBodyPost) {
+    return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1149)
+  }
+
+  @Override
+  public ResponseEntity<AcspMembership> getAcspMembershipForAcspId(
+      @NotNull String xRequestId, @Pattern(regexp = "^[a-zA-Z0-9]*$") String id) {
+    return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1146)
+  }
+
+  @Override
+  public ResponseEntity<List<AcspMembership>> getAcspMembershipForUserId(
+      final String xRequestId, final String ericIdentity, final Boolean includeRemoved) {
+    final boolean excludeRemoved = Objects.isNull(includeRemoved) || !includeRemoved;
+    return new ResponseEntity<>(
+        acspMembersService.fetchAcspMemberships(ericIdentity, excludeRemoved), HttpStatus.OK);
```

Comment:
Logging?
we need debug logging.

think about if we need any validation checks.

what should happen if we get more than one active acsp for a user, should we throw an error saying more then one active acsp exists. or should we just return.

think about it.

Reply:
Will give more of a think to the multiple 'active' acsp thing.

Reply:
Will there ever be two active acsps for a given user? Seems that detail depends on how the other endpoints are designed/implemented.  It would be easy to throw an exception if there are more than one active, but not sure how this would go with the front-end integration.

Reply:
there should not be 2 active acsps for a given user.

Reply:
I get that, but what I mean is - is this really the place to handle that? It seems if it gets to this point and we already have two active acsps, then something has went wrong with the other endpoint integrations.   Sure, we could add logs to make us aware of it given this somehow happens, but the root cause of the issue would be elsewhere, if that makes sense?